src/main/java/com/configcat/ConfigCatLogMessages.java

```diff
@@ -0,0 +1,254 @@
+package com.configcat;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+final class ConfigCatLogMessages {
```

Comment:
While this refactor is technically correct and definitely an improvement (as it reduces redundancy and makes the code clearer semantically), I think we can do even better: instead of just extracting the message text/message formatting logic, the newly introduced helper methods could do the actual logging, by means of which these methods would also **encapsulate the corresponding log level and event id**. Which seems advisable **because these parameters go together**.

So instead of having
```java
String errorMessage = ConfigCatLogMessages.getConfigJsonIsNotPresentedWitDefaultValue("defaultValue", defaultValue);
this.logger.error(1000, errorMessage);
this.configCatHooks.invokeOnFlagEvaluated(EvaluationDetails.fromError(key, defaultValue, errorMessage, user));
```

I'd prefer something like this:
```java
String errorMessage = this.logger.configJsonIsNotPresentedWitDefaultValue("defaultValue", defaultValue);
this.configCatHooks.invokeOnFlagEvaluated(EvaluationDetails.fromError(key, defaultValue, errorMessage, user));
```

AFAIK, Java doesn't have extension methods, which means that this can only be achieved by adding the helper methods to the internal logger class (`ConfigCatLogger`). But I think that's fine, it's exactly how this was implemented in the JS SDK (see https://github.com/configcat/common-js/blob/b66757bf1aa180d9a66c292a66b210d8a528dddc/src/ConfigCatLogger.ts#L170-L177).

Of course, this is just a suggestion. The current solution is already ok. 👍